### ISSUE TITLE
error occured when run catkin deps fetch

### ISSUE THREAD

**Issue by sunbin1205** [PRBonn/fast_change_detection#1]
thanks for your share!
error describe like title

INFO:deps: Will print status messages while cloning.
INFO:deps: Using 4 threads.
INFO:deps: Avoid fetching ROS packages.
INFO:deps: [ROS]: Searching all packages.
INFO:deps: [ROS]: Not found. Ignoring pre-defined ROS packages.
INFO:deps: Searching for dependencies.
 [catkin]                 : Found 3 valid dependencies                
 [fast_change_detection]  : Found 1 valid dependencies                
INFO:deps: Package [glow]: Skip default urls. Explicit one defined: https://github.com/jbehley/glow.git
 [glow]                   : Found 0 valid dependencies                
INFO:deps: Checking merged dependencies:
 [python-argparse]        : [NOT FOUND]                               
 [python-empy]            : [NOT FOUND]                               
 [python-catkin-pkg]      : [NOT FOUND]                               
 [glow]                   : [FOUND]: https://github.com/jbehley/glow.git
INFO:deps: Cloning valid dependencies:
 [glow]                   : [ALREADY EXISTS]                 
INFO:deps: New packages available. Process their dependencies now.
INFO:deps: Searching for dependencies.
INFO:deps: No new dependencies. Done.
INFO:deps: Pulling packages:
 [glow]                   : [GIT ERROR]  

hope your reply!thanks

**Comment by Eman7C7** [comment]
I just tried to reproduce this on a clean Ubuntu 16.04 install. I get the same error, but the program compiles correctly anyway. Try to compile it with `catkin build fast_change_detection` and let me know if it works. If not, please post the output of that command.

**Comment by sunbin1205** [comment]
I am very glad to receive your reply, but unfortunately I have a new problem!
The project can build successfully, but error occurs when I try to run the example under bin.

./fastcd_example ./dataset/container/
QXcbIntegration: Cannot create platform OpenGL context, neither GLX nor EGL are enabled
QXcbIntegration: Cannot create platform OpenGL context, neither GLX nor EGL are enabled
OpenGL Context Version 5.0 core profile
Missing GL version
terminate called after throwing an instance of 'std::runtime_error'
  what():  Failed to initialize GLEW.
已放弃 (核心已转储)
what's wrong？hope for your help！thanks

**Comment by sunbin1205** [comment]
I am very glad to receive your reply! The project build was successful, but running example showed some errors.
QXcbIntegration: Cannot create platform OpenGL context, neither GLX nor EGL are enabled
QXcbIntegration: Cannot create platform OpenGL context, neither GLX nor EGL are enabled
OpenGL Context Version 5.0 core profile
Missing GL version
terminate called after throwing an instance of 'std::runtime_error'
  what():  Failed to initialize GLEW.
已放弃 (核心已转储)
Is it because the opencl I installed is version 1.2? But I haven't found the version 3.3 above, can you share your opencl installation download link?

Number of platforms                               1
  Platform Name                                   Intel(R) OpenCL
  Platform Vendor                                 Intel(R) Corporation
  Platform Version                                OpenCL 1.2 LINUX
  Platform Profile                                FULL_PROFILE
......
what is wrong?hope for your help!thanks

**Comment by Eman7C7** [comment]
The library does not require OpenCL, just OpenGL.
It seems your system has not 3D acceleration. Are you trying to run it on a virtual machine?

**Comment by stevenczwu** [comment]
@Eman7C7 Hey there. I am trying to run it on a virtual machine, and I got error as this:
**Unable to handle package.xml format version '3'**

I tried with **sudo apt-get update** && **sudo apt-get install --only-upgrade python-catkin-pkg**, but it is still not working... do you have any idea? Thanks!

**Comment by CESI2Jaafar** [comment]
Hi  @Eman7C7,
I have tried many times in google colab with proper packages. I have tried with Ubuntu 16.04 and its proper ROS. Still I get this error: If you give me a bit guidance or comment, it would be highly appreciated: 
root@jnajafirad:~/catkin_ws/src# catkin init
Catkin workspace `/root/catkin_ws` is already initialized. No action taken.
----------------------------------------------------
Profile:                     default
Extending:                   None
Workspace:                   /root/catkin_ws
----------------------------------------------------
Build Space:        [exists] /root/catkin_ws/build
Devel Space:        [exists] /root/catkin_ws/devel
Install Space:      [unused] /root/catkin_ws/install
Log Space:          [exists] /root/catkin_ws/logs
Source Space:       [exists] /root/catkin_ws/src
DESTDIR:            [unused] None
----------------------------------------------------
Devel Space Layout:          linked
Install Space Layout:        None
----------------------------------------------------
Additional CMake Args:       None
Additional Make Args:        None
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
----------------------------------------------------
Buildlisted Packages:        None
Skiplisted Packages:         None
----------------------------------------------------


----------------------------------------------------
WARNING: Your workspace is not extending any other
result space, but it is set to use a `linked` devel
space layout. This requires the `catkin` CMake
package in your source space in order to be built.
